File: app/models/customer.py
# app/models/customer.py
from app.extensions import db
from app.models.base import TimestampMixin, UUIDPrimaryKeyMixin


class Customer(UUIDPrimaryKeyMixin, TimestampMixin, db.Model):
    __tablename__ = "customers"
    __table_args__ = (db.UniqueConstraint("tenant_id", "customer_number", name="uq_customers_tenant_customer_number"),)

    tenant_id = db.Column(db.String(36), db.ForeignKey("tenants.id"), nullable=False, index=True)
    customer_number = db.Column(db.String(50), nullable=False)
    first_name = db.Column(db.String(100), nullable=False)
    last_name = db.Column(db.String(100))
    other_names = db.Column(db.String(150))
    phone = db.Column(db.String(50), index=True)
    email = db.Column(db.String(150), index=True)
    gender = db.Column(db.String(20))
    date_of_birth = db.Column(db.Date)
    id_number = db.Column(db.String(50))
    address = db.Column(db.Text)
    city = db.Column(db.String(100))
    loyalty_points = db.Column(db.Numeric(18, 2), nullable=False, default=0)
    is_active = db.Column(db.Boolean, nullable=False, default=True)

File: frontend/src/types/entities/customer.ts
/**
 * ============================================================================
 * Hela360 Customer Entity
 * ============================================================================
 *
 * Canonical frontend representation of the backend Customer serializer.
 *
 * The current Customer service returns backend JSON directly, so this contract
 * intentionally uses snake_case field names.
 *
 * ============================================================================
 */

export interface Customer {
  id: string;
  tenant_id: string;
  customer_number: string;
  first_name: string;
  last_name: string | null;
  other_names: string | null;
  full_name: string;
  phone: string | null;
  email: string | null;
  gender: string | null;
  date_of_birth: string | null;
  id_number: string | null;
  address: string | null;
  city: string | null;
  loyalty_points: string;
  is_active: boolean;
  created_at: string | null;
  updated_at: string | null;
}
